sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsPartitions.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connector.catalog;
+
+import java.util.Map;
+
+import org.apache.spark.annotation.Experimental;
+
+/**
+ * Catalog methods for working with Partitions.
+ */
+@Experimental
+public interface SupportsPartitions extends TableCatalog {
+
+    /**
+     * Create partitions in an existing table, assuming it exists.
+     *
+     * @param ident a table identifier
+     * @param partitions transforms to use for partitioning data in the table
+     * @param ignoreIfExists
+     */
+    void createPartitions(
+            Identifier ident,
+            TablePartition[] partitions,
+            Boolean ignoreIfExists);
+
+    /**
+     * Drop partitions from a table, assuming they exist.
+     *
+     * @param ident a table identifier
+     * @param partitions a list of string map for existing partitions
+     * @param ignoreIfNotExists
+     */
+    void dropPartitions(
+            Identifier ident,
+            Map<String, String>[] partitions,
+            Boolean ignoreIfNotExists);
+
+    /**
+     * Override the specs of one or many existing table partitions, assuming they exist.
+     *
+     * @param ident a table identifier
+     * @param oldpartitions a list of string map for existing partitions to be renamed
+     * @param newPartitions a list of string map for new partitions
+     */
+    void renamePartitions(
+            Identifier ident,
+            Map<String, String>[] oldpartitions,
+            Map<String, String>[] newPartitions);
```

Comment:
Can you give an example of what this does?

Reply:
It happens when we want to rename partition identifier and don't want to write again.
A lot of Examples for this. Such as, we have write a FileTable partition `test`, the location for this is `/warehouse/table/part=test/xxxx`. This will be used when we want rename partition `test` to `actual`.

Reply:
Doesn't that modify the data? Would it cause data files to be moved from one partition to another? How is this called from SQL?

I'm not sure that this is an operation that should be generally supported if it is a form of `UPDATE` that requires the modified columns are partition columns. It seems like this operation is very specific to Hive tables.

Reply:
Actually `RENAME PARTITION` is widely used in mysql/postgres/oracle or any other sources, some of them rename partition identifier while others will modify user data.
Thus, AFSIT, it should be supported in common `Partition API`s.

Reply:
I commented on the other thread. I think that the partitions that can be renamed are slightly different than the ones that Hive exposes, so it may not be appropriate for this API.

To support operations like this, we might want to expand the `PARTITIONED BY` expression in `CREATE TABLE` and have a separate API for Postgres/Oracle style partitions.